### PR TITLE
[HYDRA-685] Twitter Spark Streaming Source Plugin

### PR DESCRIPTION
**Pull request by knazir**
Implementation of a Twitter spark streaming source. 
I don't think a unit test is easily possible, if anyone has an idea please let me know.

Issue: https://issues.cask.co/browse/HYDRA-685
Build: http://builds.cask.co/browse/HYP-BAD61-6

### CODE REVIEW

File: spark-plugins/pom.xml [hunk 1]
```diff
@@ -165,7 +177,7 @@
         <configuration>
           <instructions>
             <_exportcontents>
-              co.cask.hydrator.plugin.batch.spark.*;org.apache.spark.streaming.kafka.*;org.apache.spark.streaming.scheduler.*;kafka.serializer.*
+              co.cask.hydrator.plugin.batch.spark.*;org.apache.spark.streaming.kafka.*;org.apache.spark.streaming.scheduler.*;kafka.serializer.*;org.apache.spark.streaming.twitter.*;twitter4j.*; twitter4j.auth.*;
```
Comment (anwar6953) [inline spark-plugins/pom.xml]: you added 3 exports. Are all 3 necessary?

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    );
+  }
+
+  private StructuredRecord convertTweet(Status tweet) {
+    StructuredRecord.Builder recordBuilder = StructuredRecord.builder(SCHEMA);
+    recordBuilder.set(ID, tweet.getId());
+    recordBuilder.set(MSG, tweet.getText());
+    recordBuilder.set(LANG, tweet.getLang());
+    Date tweetDate = tweet.getCreatedAt();
+    if (tweetDate != null) {
```
Comment (anwar6953) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java:115]: how are you determining what's nullable and what's not?

File: spark-plugins/pom.xml (hunk 1 above)
Comment (knazir) [inline spark-plugins/pom.xml]: Ah you're right, `twitter4j.*` was not necessary, the other two are though.

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java (hunk 2 above)
Comment (knazir) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java:115]: The logic is taken from the deprecated `TwitterSource` in realtime sources. The history says that @albertshau was the last to touch those lines, maybe he knows? I tried looking in the implementation of `StatusJSONImpl` but couldn't determine which fields could be nullable.

File: spark-plugins/pom.xml (hunk 1 above)
Comment (knazir) [inline spark-plugins/pom.xml]: Nope, just kidding, all 3 are necessary my bad.

Comment (albertshau) [inline spark-plugins/pom.xml]: if you have twitter4j.\* then I don't think you need twitter4j.auth.*

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java (hunk 2 above)
Comment (albertshau) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java:115]: I think it was just from looking at the data? don't really remember

**Comment by albertshau** [comment]
just one comment otherwise lgtm

File: spark-plugins/pom.xml (hunk 1 above)
Comment (knazir) [inline spark-plugins/pom.xml]: Ah my bad, thought it was like importing packages. Just tried it and you're right.

**Comment by knazir** [comment]
Guild is breen: http://builds.cask.co/browse/HYP-BAD61-10/commit squashing and merging.

**Pull request merged**